e2e-pw/src/oss/poms/saved-views.ts

```diff
@@ -0,0 +1,406 @@
+    await this.clickDeleteBtn();
+  }
+
+  async editView(
+    name: string = updatedView.name,
+    description: string = updatedView.description,
+    color: Color = "Blue",
+    newColor: Color = updatedView.color
+  ) {
+    await this.nameInput().clear();
+    await this.nameInput().type(name);
+    await this.descriptionInput().clear();
+    await this.descriptionInput().type(description);
+    await this.colorInput(color).click();
+    await this.colorOption(newColor).click();
+
+    await this.saveButton().click();
+  }
+
+  async clickColor(color: Color = "Gray") {
+    await this.colorInput(color).click();
+  }
+
+  async clearView() {
+    // await this.clearViewBtn().waitFor({ state: "visible" });
+    await this.clearViewBtn().click();
+  }
+
+  async clickCloseModal() {
+    await this.closeModalBtn().click();
+  }
+
+  clearViewBtn() {
+    return this.locator.getByTestId("btn-selection-clear");
+  }
+
+  closeModalBtn() {
+    return this.dialogLocator.getByTestId("btn-close");
+  }
+
+  saveNewViewBtn() {
+    return this.locator.getByTestId("create-new");
+  }
+
+  unsavedView() {
+    return this.locator.getByRole("button", {
+      name: "Unsaved view",
+      exact: true,
+    });
+  }
+
+  async unsavedViewCount() {
+    return await this.unsavedView().count();
+  }
+
+  async openSelect() {
+    await this.unsavedView().click();
+  }
+
+  async openCreateModal() {
+    await this.openSelect();
+    await this.saveNewViewBtn().click();
+  }
+
+  savedView(name: string = "test") {
+    return this.locator.getByRole("button", { name });
+  }
+
+  async savedViewCount(name: string) {
+    return await this.locator.getByRole("button", { name }).count();
+  }
+
+  savedViewOption(name: string = "test") {
+    return this.locator.getByRole("option", { name, exact: true }).first();
+  }
+
+  async savedViewOptionCount(name: string = "test") {
+    return await this.locator
+      .getByRole("option", { name, exact: true })
+      .count();
+  }
+
+  nameInput() {
+    return this.dialogLocator.getByTestId("input-name");
+  }
+
+  descriptionInput() {
+    return this.dialogLocator.getByTestId("input-description");
+  }
+
+  colorInput(c: Color = "Gray") {
+    return this.dialogLocator.getByRole("button", { name: c });
+  }
+
+  colorOption(c: Color = "Purple") {
+    return this.dialogLocator.getByRole("option", { name: c });
+  }
+
+  saveButton() {
+    return this.dialogLocator.getByRole("button", {
+      name: "Save view",
+      exact: true,
+    });
+  }
+
+  cancelButton() {
+    return this.dialogLocator.getByRole("button", {
+      name: "Cancel",
+      exact: true,
+    });
+  }
+
+  colorListContainer() {
+    return this.dialogLocator.getByRole("listbox");
+  }
+
+  nameError() {
+    return this.dialogLocator.getByText("Name already exists");
+  }
+
+  searchInput() {
+    return this.locator
+      .getByTestId("search-container")
+      .getByTestId("search-input");
+  }
+
+  deleteBtn() {
```

Comment:
You could make these functions that return `Locator`s private and make the functions that perform operations public (`clickDeleteBtn`, in this case)